src/components/inputs/multi-select/multi-select.component.tsx

```diff
@@ -59,6 +61,30 @@ const MultiSelect: React.FC<FormFieldProps> = ({ question, onChange, handler, pr
     }
   }, [previousValue]);
 
+  useEffect(() => {
+    if (isFirstRender.current && counter === 1) {
+      setInitiallyCheckedQuestionItems(initiallySelectedQuestionItems.map((item) => item.concept));
+      isFirstRender.current = false;
+    }
+  }, [initiallySelectedQuestionItems, isFirstRender ]);
```

Comment:
The use of `isFirstRender` and `initiallyCheckedQuestionItems` is still a little confusing to me. By the look of things, you're using it to track checked items but those are already tracked with `initiallySelectedQuestionItems` is there a catch with reusing the later?

Reply:
Since the field value keeps changing(its a multi select), the value of `initiallySelectedQuestionItems` keeps getting reset on each rerender, thats why i had to find a way to track it